validate_key.py

```diff
@@ -0,0 +1,56 @@
+from datetime import datetime, timedelta
+import hashlib
+
+from pywell.entry_points import run_from_cli, run_from_api_gateway
+
+
+DESCRIPTION = 'Validate a download key.'
+
+ARG_DEFINITIONS = {
+    'KEY': 'Key to validate.',
+    'MAX_AGE': 'Number of days a key should be considered valid.',
+    'SECRET': 'Secret to use for validation.'
+}
+
+REQUIRED_ARGS = ['KEY', 'MAX_AGE', 'SECRET']
+
+
+def main(args):
+    # If key doesn't have enough parts, it's invalid.
+    if len(args.KEY.split('.')) != 5:
+        return {'valid': False}
+    [key_created, age, source, campaign, hash] = args.KEY.split('.')
+    m = hashlib.sha256()
```

Comment:
if you are validating a key, then you should use an HMAC construction (rather than a sha256 hash).  We use sha hackily in places to support ActionKit, but is this related to actionkit akids?

Reply:
I think we resolved this in Slack, but to document here: to allow these keys to be created via queries, we're somewhat limited in the hash options.